src/test/java/com/cloudbees/plugins/credentials/casc/CredentialsProviderTest.java

```diff
@@ -0,0 +1,141 @@
+
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.UsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.domains.DomainCredentials;
+import com.cloudbees.plugins.credentials.domains.HostnameRequirement;
+import com.cloudbees.plugins.credentials.impl.DummyCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.ItemGroup;
+import hudson.security.ACL;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import org.acegisecurity.Authentication;
+import org.jenkinsci.Symbol;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.TestExtension;
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.StringContains.containsString;
+
+public class CredentialsProviderTest {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("CredentialsProviderExtension.yaml")
+    public void import_credentials_provider_extension_credentials() {
+        List<DummyCredentials> dummyCred = CredentialsProvider.lookupCredentials(
+                DummyCredentials.class, j.jenkins, ACL.SYSTEM,
+                Collections.emptyList()
+        );
+        assertThat(dummyCred, hasSize(1));
+        assertThat(dummyCred.get(0).getUsername(), equalTo("user1"));
+
+        // the system provider works fine too
+        List<UsernamePasswordCredentials> ups = CredentialsProvider.lookupCredentials(
+                UsernamePasswordCredentials.class, j.jenkins, ACL.SYSTEM,
+                Collections.singletonList(new HostnameRequirement("api.test.com"))
+        );
+        Assert.assertThat(ups, hasSize(1));
+        final UsernamePasswordCredentials up = ups.get(0);
+        Assert.assertThat(up.getPassword().getPlainText(), equalTo("password"));
```

Comment:
could we use has plain text? https://github.com/jenkinsci/jenkins-test-harness/blob/master/src/main/java/org/jvnet/hudson/test/JenkinsMatchers.java#L404

```suggestion
       assertThat(up.getPassword(), hasPlainText("password"));
```